### PR TITLE
Update phpspec

### PR DESCRIPTION
**Pull request by Big-Shark**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | yes
| Related tickets | phpspec/phpspec#1274
| Documentation   | no
| License         | MIT

#### What's in this PR?

Just update phpspec version

### CODE REVIEW

**Comment by dbu** [comment]
thanks! looking at the lowest version build failure, i think we need a higher minimum version of either the coverage thing or phpspec itself

File: composer.json [hunk 1]
```diff
@@ -21,8 +21,8 @@
         "php-http/promise": "^1.0"
     },
     "require-dev": {
-        "phpspec/phpspec": "^2.4",
-        "henrikbjorn/phpspec-code-coverage" : "^1.0"
+        "phpspec/phpspec": "^4.0|^5.0|^6.0",
```
Comment (GrahamCampbell) [inline composer.json]: phpspec 4 can be dropped?

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (Big-Shark) [reply composer.json]: @GrahamCampbell In the current moment httplug support php version 7.0, but phpspec 5.0 support only 7.1, and I added phpspec 4.0 for php 7.0

Comment (GrahamCampbell) [reply composer.json]: Oh, for some reason I though this package was PHP 7.1+.

In which case, we should change it to `^4.2` probably?

Comment (Big-Shark) [reply composer.json]: @GrahamCampbell Why not ^4.3?

Comment (dbu) [reply composer.json]: for testing tools, i am happy with the most strict restrictions that still allow to build, if ^4.3 works thats better than ^4.2. we don't give anybody more flexibility if we allow 4.2

**Comment by dbu** [comment]
can you please try phpstan 4.3.4? seems 4.3.0 still has that configuration bug

File: composer.json [hunk 2]
```diff
@@ -21,8 +21,8 @@
         "php-http/promise": "^1.0"
     },
     "require-dev": {
-        "phpspec/phpspec": "^2.4",
-        "henrikbjorn/phpspec-code-coverage" : "^1.0"
+        "phpspec/phpspec": "^4.3|^5.0|^6.0",
+        "friends-of-phpspec/phpspec-code-coverage" : "^4.1"
```
Comment (dbu) [inline composer.json:25]: use ^4.1.2 of this, to avoid the lowest-version failure?

**Comment by Big-Shark** [comment]
@dbu This is strange, this is not best solution, but this is work.

**Review by dbu — APPROVED** [review]
oh, i did not realize we had coverage on the lowest version build only... yeah i agree with your solution, don't see why we would want coverage on lowest version build.

**Comment by dbu** [comment]
thanks a lot!

**Pull request merged**